glean/src/server_events.rs

```diff
@@ -0,0 +1,308 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Required imports
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+// log type string used to identify logs to process in the Moz Data Pipeline
+const GLEAN_EVENT_MOZLOG_TYPE: &str = "glean-server-event";
+
+// Code below is static, regardless of what is defined in `metrics.yaml`:
+pub struct GleanEventsLogger {
+    // Application Id to identify application per Glean standards
+    pub app_id: String,
+    // Version of application emitting the event
+    pub app_display_version: String,
+    // Channel to differentiate logs from prod/beta/staging/devel
+    pub app_channel: String,
+}
+
+// Exported type for public method parameters
+// Default impl empty values will be omitted in json from ping struct definition
```

Comment:
FWIW: serde offers a flag for this: https://serde.rs/field-attrs.html#skip_serializing_if